skyportal/handlers/api/annotation.py

```diff
@@ -232,90 +232,94 @@ def post(self, associated_resource_type, resource_id):
+                        f'Could not access spectrum {spectrum_id}.', status=403
+                    )
+                data['spectrum_id'] = spectrum_id
+                data['obj_id'] = spectrum.obj_id
+                schema = AnnotationOnSpectrum.__schema__(exclude=["author_id"])
+                try:
+                    schema.load(data)
+                except ValidationError as e:
+                    return self.error(
+                        f'Invalid/missing parameters: {e.normalized_messages()}'
+                    )
+
+                annotation = AnnotationOnSpectrum(
+                    data=annotation_data,
+                    spectrum_id=spectrum_id,
+                    obj_id=spectrum.obj_id,
+                    origin=origin,
+                    author=author,
+                    groups=groups,
                 )
-            data['spectrum_id'] = spectrum_id
-            data['obj_id'] = spectrum.obj_id
-            schema = AnnotationOnSpectrum.__schema__(exclude=["author_id"])
-            try:
-                schema.load(data)
-            except ValidationError as e:
+            else:
                 return self.error(
-                    f'Invalid/missing parameters: {e.normalized_messages()}'
+                    f'Unknown resource type "{associated_resource_type}".'
                 )
 
-            annotation = AnnotationOnSpectrum(
-                data=annotation_data,
-                spectrum_id=spectrum_id,
-                obj_id=spectrum.obj_id,
-                origin=origin,
-                author=author,
-                groups=groups,
-            )
-        else:
-            return self.error(f'Unknown resource type "{associated_resource_type}".')
-
-        DBSession().add(annotation)
-        try:
-            self.verify_and_commit()
-        except IntegrityError as e:
-            return self.error(f'Annotation already exists: {str(e)}')
+            session.add(annotation)
 
-        if isinstance(
-            annotation, (Annotation, AnnotationOnSpectrum)
-        ):  # annotation on object or object related data
-            self.push_all(
-                action='skyportal/REFRESH_SOURCE',
-                payload={'obj_key': annotation.obj.internal_key},
-            )
-        if isinstance(annotation, AnnotationOnSpectrum):
-            self.push_all(
-                action='skyportal/REFRESH_SOURCE_SPECTRA',
-                payload={'obj_internal_key': annotation.obj.internal_key},
-            )
-        return self.success(data={'annotation_id': annotation.id})
+            try:
+                session.commit()
+            except IntegrityError as e:
+                return self.error(f'Annotation already exists: {str(e)}')
+
+            if isinstance(
+                annotation, (Annotation, AnnotationOnSpectrum)
+            ):  # annotation on object or object related data
+                self.push_all(
+                    action='skyportal/REFRESH_SOURCE',
```

Comment:
Do we need both these websocket messages for `AnnotationOnSpectrum`?

Reply:
Yes because it updates both the annotation list on the source page and the once on the "manage data" form.